src/code_1A5830.c

```diff
@@ -98,7 +98,31 @@ ApiStatus ResumeTakeTurn(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "code_1A5830", BindIdle);
+ApiStatus BindIdle(ScriptInstance* script, s32 isInitialCall) {
```

Comment:
A few things with this function:

I'd call var0 `idleCode` and give it the type of whatever `actor->idleCode` is. Also, when you check if actor->idleScript != 0 and then set it to 0, please use `NULL` instead for both. We like to use NULL for pointer types rather than 0.